### PR TITLE
feat: Add new constraint operators

### PR DESCRIPTION
**Pull request by chriswk**
- Adds String operators, startsWith, endsWith, contains
  with support for case insentivity if desired

- Adds Number operators LT, LTE, EQ, GTE, GT for comparing numbers

- Adds Date operators for supporting DATE_AFTER and DATE_BEFORE

- Adds Semver operators for support SEMVER_LT, SEMVER_EQ, SEMVER_GT

### CODE REVIEW

**Comment by chriswk** [comment]
[Java SDK - Add support for new constraint operators](https://trello.com/c/tje5YbwF/833-java-sdk-add-support-for-new-constraint-operators)

**Comment by chriswk** [comment]
So, I downloaded the PR for client-specification and managed to run all tests specified green, once I spotted that we did indeed extend the context to include a currentDate field and the Constraint to have both values and value fields.

File: src/test/java/io/getunleash/RunOnJavaVersionsCondition.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+public class RunOnJavaVersionsCondition implements ExecutionCondition {
+    @Override
+    public ConditionEvaluationResult evaluateExecutionCondition(ExtensionContext extensionContext) {
+        Optional<RunOnJavaVersions> annotation =
+                findAnnotation(extensionContext.getElement(), RunOnJavaVersions.class);
+        return annotation
+                .map(
+                        a -> {
+                            String runtimeVersion = getJavaVersion();
```
Comment (sighphyre) [inline src/test/java/io/getunleash/RunOnJavaVersionsCondition.java:19]: This is cool!

**Review by sighphyre — APPROVED** [review]
Awesome!

**Pull request merged**